### ISSUE TITLE
Cannot input text in minibuffer

### ISSUE THREAD

**Issue by molekular** [atlas-engineer/next#772]
### Describe situation
I open next, question "restore previous session" appears, I cannot type anything in the minibuffer. "Return" to confirm works, I can paste text into the minibuffer, key-bindings work too, but no text input. I tried starting with --no-init, same problem, so it should not be misconfigured in the init-file.

### System Information
Archlinux
EXWM window manager version 0.24
Emacs 26.03
next Version 1.5.0-f027c473
webkit2gtk-2.28.2-2

### Additional information from --verbose
```
<DEBUG> [17:25:18] next browser.lisp (initialize-modes buffer) -
  NEXT::MODE-CLASS: MINIBUFFER-MODE BUFFER: #<MINIBUFFER {1008AF3063}>
  (NEXT::MODE-COMMAND NEXT::MODE-CLASS): #<NEXT::COMMAND {10036416D3}>
<DEBUG> [17:25:18] next minibuffer.lisp (minibuffer-mode form-fun-15) -
  Calling command MINIBUFFER-MODE.
<DEBUG> [17:25:18] next minibuffer.lisp (minibuffer-mode form-fun-15) -
  MINIBUFFER-MODE enabled.
<DEBUG> [17:25:20] next renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NEXT::KEY-STRING: "y" NEXT::KEYCODE: 52 CHARACTER: #\y NEXT::KEYVAL-NAME: "y"
<DEBUG> [17:25:20] next renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NEXT::KEY-STRING: "e" NEXT::KEYCODE: 26 CHARACTER: #\e NEXT::KEYVAL-NAME: "e"
<DEBUG> [17:25:21] next renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NEXT::KEY-STRING: "s" NEXT::KEYCODE: 39 CHARACTER: #\s NEXT::KEYVAL-NAME: "s"
<DEBUG> [17:25:24] next renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
<DEBUG> [17:25:26] next renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NEXT::KEY-STRING: NIL NEXT::KEYCODE: 37 CHARACTER: #\Nul
  NEXT::KEYVAL-NAME: "Control_L"
```
When I started using next some weeks ago I encountered the same problem - as some other user reported in [# 645](https://github.com/atlas-engineer/next/issues/645) it was caused by the code block
```
  export XMODIFIERS=@im=exwm-xim
  export GTK_IM_MODULE=xim
  export QT_IM_MODULE=xim
  export CLUTTER_IM_MODULE=xim
```
in my .xinitrc. Removing it solved the issue.

Now I just changed from vanilla emacs to doom and encountered this problem again - I don't see why. Any help would be much appreciated. Thank you for this great browser!

**Comment by Ambrevar** [comment]
Could it be that non-modifier, non-special keys are all intercepted by
Doom?
Could you try again without Doom?

**Comment by molekular** [comment]
Thank you for your quick response. I disabled Doom and it works again. I had tried with line-mode and char-mode, thinking that at least in char-mode keys should not be intercepted by emacs. 
I now did the same with my second laptop (UMPC), that has the same configuration (arch, exwm, doom, next), and there the minibuffer-input works. So, I could compare - if you have a hint what to look for I could do that.

**Comment by Ambrevar** [comment]
1. Check the program versions.  (Emacs, EXWM, Doom, Next, WebKitGTK, GTK)

2. Compare the environments (the result of the `env` command).

3. Compare your Emacs / EXWM / Doom configurations.

4. Compare your desktop init files like ~/.xinitrc, ~/.xprofile, ~/.xsession.

**Comment by molekular** [comment]
Great, thank you! I already found remains of "exwm-xim" in env on the machine where the input does not work, I think there is something wrongly configured. Other configuration files are almost identical, I will check them later. Probably I got my input method messed up, nothing to do with next. I will update here and close when fixed. Thank you!

**Comment by Ambrevar** [comment]
Did you have the same problem with other applications?

**Comment by molekular** [comment]
No, everything else works fine.

**Comment by Ambrevar** [comment]
Strange, then there must be something wrong with Next.

**Comment by molekular** [comment]
If I can provide any more information or logs or anything you want me to try ... Please let me know. I can live without doom but got used to next and don't want to miss it. Would it make sense to ask in the doom-discord-group? I saw next has been mentioned there a couple of times.

**Comment by Ambrevar** [comment]
Sure, if Doom does something with an input engine, they might know better.

**Comment by polaris64** [comment]
I encountered the same problem, however I'm not using EXWM. I'm using Ubuntu 20.04 with i3. I noticed that when I had `GTK_IM_MODULE` set to `xim` I couldn't type anything into the minibuffer in Next. Un-setting this environment variable when launching Next fixed the problem.

I looked at the logs for both variations when starting to type "hello" into the minibuffer.

With `GTK_IM_MODULE=xim`
```
** Message: 16:50:57.377: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (43, 'h', @as [], -1.0, -1.0, 104, '1')
** Message: 16:50:57.429: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (43, 'h', ['R'], -1.0, -1.0, 104, '1')
** Message: 16:50:58.177: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (26, 'e', @as [], -1.0, -1.0, 101, '1')
** Message: 16:50:58.241: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (26, 'e', ['R'], -1.0, -1.0, 101, '1')
```

Without `GTK_IM_MODULE`
```
** Message: 16:49:25.536: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (43, 'h', @as [], -1.0, -2.0, 104, '1')
** Message: 16:49:25.550: Method name: window_active
** Message: 16:49:25.550: Method parameter(s): 1
** Message: 16:49:25.580: Method name: window_active
** Message: 16:49:25.580: Method parameter(s): 1
** Message: 16:49:25.585: Method name: minibuffer_evaluate_javascript
** Message: 16:49:25.585: Method parameter(s): window id 1, javascript (...)
** Message: 16:49:25.585: Method result(s): callback id 69
** Message: 16:49:25.587: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 69)
** Message: 16:49:25.599: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (43, 'h', ['R'], -1.0, -1.0, 104, '1')
** Message: 16:49:28.588: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (26, 'e', @as [], -1.0, -2.0, 101, '1')
** Message: 16:49:28.603: Method name: window_active
** Message: 16:49:28.603: Method parameter(s): 1
** Message: 16:49:28.624: Method name: window_active
** Message: 16:49:28.624: Method parameter(s): 1
** Message: 16:49:28.632: Method name: minibuffer_evaluate_javascript
** Message: 16:49:28.632: Method parameter(s): window id 1, javascript (...)
** Message: 16:49:28.632: Method result(s): callback id 70
** Message: 16:49:28.634: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 70)
** Message: 16:49:28.659: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (26, 'e', ['R'], -1.0, -1.0, 101, '1')
```

It appears that with `xim` the `push_input_event` calls are not handled (no further logging after the event). Also the `y` parameter with `xim` is always `-1.0` whereas without `xim` it is `-2.0` in the initial event and `-1.0` in the second event. Perhaps this has something to do with it, e.g. perhaps events are only handled when `y == -2.0`?

**Comment by jmercouris** [comment]
I'm sorry, I have no idea why you are experiencing this issue. I don't remember of a check for -1 or -2 specifically.

Importantly we have improved Nyxt very substantially since the last release. Could you please try compiling from source and see if you can still recreate the problem?

**Comment by polaris64** [comment]
I've just built Nyxt from source (current HEAD of master, 799836b660ce76e98d79aae20ac9a7888607d657) and the problem unfortunately still persists.

Here is the log for the same test when using `GTK_IM_MODULE=xim`: -
```
<DEBUG> [21:29:34] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "h" NYXT::KEYCODE: 43 CHARACTER: #\h NYXT::KEYVAL-NAME: "h"
<DEBUG> [21:29:36] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "e" NYXT::KEYCODE: 26 CHARACTER: #\e NYXT::KEYVAL-NAME: "e"
```

Here's the equivalent when `GTK_IM_MODULE` is unset (I have cut out some lines for brevity): -
```
<DEBUG> [21:34:39] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "h" NYXT::KEYCODE: 43 CHARACTER: #\h NYXT::KEYVAL-NAME: "h"
<DEBUG> [21:34:39] nyxt input.lisp (dispatch-input-event) -
  Insert "h" in minibuffer
<DEBUG> [21:34:39] nyxt fuzzy.lisp (fuzzy-match) -
  ((help (f1 f1) 0.083333336) (enable-hook-handler 0.05263158)
   (delete-history-entry 0.05) (show-bookmarks (c-b) 0.05)
   (delete-other-buffers 0.05) (switch-buffer-domain 0.05)
   (disable-hook-handler 0.05) (help-mode [help-mode] 0.04761905)
---8<--- SNIP ---8<---
<DEBUG> [21:34:42] nyxt renderer-gtk.lisp (on-signal-key-press-event gtk-window) -
  NYXT::KEY-STRING: "e" NYXT::KEYCODE: 26 CHARACTER: #\e NYXT::KEYVAL-NAME: "e"
<DEBUG> [21:34:42] nyxt input.lisp (dispatch-input-event) -
  Insert "e" in minibuffer
<DEBUG> [21:34:42] nyxt fuzzy.lisp (fuzzy-match) -
  ((help (f1 f1) 0.75) (help-mode [help-mode] 0.64285713)
   (delete-other-buffers 0.30811387) (spell-check-word [web-mode] 0.2878878)
   (spell-check-suggest-word [web-mode] 0.262491)
   (jump-to-heading (c-.) [web-mode] 0.2604167)
   (spell-check-word-at-cursor [web-mode] 0.25785777)
   (spell-check-highlighted-word [web-mode] 0.25369978)
   (zoom-out-page (c-button5, c-hyphen) [web-mode] 0.15360574))
```

**Comment by jmercouris** [comment]
That's so fascinating. It feels as if it *should* be working. I wonder why that is... thank you for the report.

**Comment by Ambrevar** [comment]
Are you saying that you pressed "e" and the letter was not inserted in
the minibuffer?

Do you see the cursor (a grey square) in the minibuffer?
If so, does this cursor advance when you press it?

Have you tried to run the following before starting Nyxt?

```
GTK_IM_MODULE=xim
XMODIFIERS=@im=ibus
ibus --daemonize --replace --xim
```

**Comment by polaris64** [comment]
Yes, when the minibuffer is open no letters at all are inserted. I see a solid grey cursor, but it doesn't move when I type any characters. Other key commands work fine though, such as `C-g` to exit the minibuffer and `C-x C-c` to quit.

I just tried your suggestion (I'm assuming you meant `ibus-daemon --daemonize --replace --xim`) but the result is the same. So far the only method I've found to get it working is to unset `GTK_IM_MODULE`.

I should note that I just tried this on Next 1.5.0, not the custom build I made yesterday. I'm not able to try that yet but I will later if possible. If there's any difference I'll let you know.
